### ISSUE TITLE
containers-storage does not read environment overrides for "local" config of podman and friends

### ISSUE THREAD

**Issue by molar** [containers/skopeo#1272]
Hi

I am trying to get skopeo to copy a set of images from various sources to a custom containers-storage directory. However it seems it does not use the CONTAINERS_STORAGE_CONF to determine which path to use when copying. This is also written in the documentation that the /etc/containers/storage.conf path is always used. Is it on the roadmap to support the same overrides as the podman tools have?

Reproduce

1. set the CONTAINERS_STORAGE_CONF environment variable to point to a custom storage configuration
2. in the custom storage configuration set the "graphroot" and "rootless_storage_path" to another directory.
3. test that it works by podman images ... podman pull ... etc..
4. use skopeo to copy from docker to the containers-storage does not use the overridden conf to determine where to put the image.
below is standing in a directory performing the above steps and pointing the storage paths to an empty directory that is then populated by podman. when using skopeo it uses the configuration found in /etc/containers/storage.conf and i guess in the case of rootless it uses the ~/.config/containers/storage.conf

```
$ podman images
REPOSITORY             TAG     IMAGE ID      CREATED      SIZE
quay.io/podman/stable  latest  30b9af9a08d6  5 hours ago  416 MB
localhost/latest       latest  9a245e58983c  6 hours ago  3.66 GB

$ skopeo copy docker://alpine containers-storage:alpine
Getting image source signatures
Copying blob 540db60ca938 [--------------------------------------] 0.0b / 0.0b
Copying config 6dbb9cc540 done
Writing manifest to image destination
Storing signatures
$ podman images
REPOSITORY             TAG     IMAGE ID      CREATED      SIZE
quay.io/podman/stable  latest  30b9af9a08d6  5 hours ago  416 MB
localhost/latest       latest  9a245e58983c  6 hours ago  3.66 GB
```

**Comment by rhatdan** [comment]
Sure, interested in opening a PR to make that happen?

**Comment by mtrmac** [comment]
It doesn’t make all that much sense to me for every single c/storage user to add environment variables. Wouldn’t that be a feature of c/storage, or at maybe a subpackage in c/common like the `login` command?

**Comment by molar** [comment]
I have made an attempt at fixing this in the `containers/storage` repo under the `types/options.go` in the `DefaultStorageOptions` function. 
I check if the environment variable is set an basically use the same function call as the podman code does. 
i will be happy to open a PR against the `containers/storage` repo

**Comment by mtrmac** [comment]
https://github.com/containers/storage/pull/890 was included by #1311 and released in Skopeo 1.4.0.  Thanks for that work!
